Build failure: calyx-vpn

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *calyx-vpn*

### Build log

https://gist.github.com/RoGreat/69d09d51c7a4dd336495de8c0cc36ddb

```
 last 10 log lines:
 > Updating '/build/source/gui/i18n/main_zh_TW.qm'...
 >     Generated 77 translation(s) (77 finished and 0 unfinished)
 >     Ignored 3 untranslated source text(s)
 > [+] Now building Qml app with Qt qmake
 > [+] Using qmake in: /nix/store/4xk5njji85yszys9hg8v2mml7wp4y8k1-qtbase-5.15.14-dev/bin/qmake
 > Info: creating stash file /build/source/build/qt/.qmake.stash
 > Project MESSAGE: [qmake] VENDOR_PATH: providers
 > Cannot open /build/source/build/qt/gui_gui_qmlcache.qrc for writing.
 > Project ERROR: Unknown module(s) in QT: quickcontrols2
 > make: *** [Makefile:156: build_gui] Error 3
```

### Additional context

Looks related to recent `riseup-vpn` update #344549

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@vigneshpai2003 
@cleeyv
@ju1m 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.9-zen1, NixOS, 24.11 (Vicuna), 24.11.20241006.c31898a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `""`
 - nixpkgs: `/nix/store/rs4fjbnw4qx7ns2hzzrz2iz52va7vs5z-source`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/networking/bitmask-vpn/default.nix
+++ b/pkgs/tools/networking/bitmask-vpn/default.nix
@@ -1,9 +1,11 @@
 { lib
 , stdenv
 , substituteAll
+, git
 , fetchFromGitLab
 , buildGoModule
 , wrapQtAppsHook
+, python3
 , python3Packages
 , pkg-config
 , openvpn
@@ -30,7 +32,8 @@ let
     owner = "leap";
     repo = "bitmask-vpn";
     rev = "8b3ac473f64b6de0262fbf945ff25af8029134f1";
-    sha256 = "sha256-nYMfO091w6H7LyY1+aYubFppg4/3GiZZm4e+0m9Gb3k=";
+    leaveDotGit = true;
+    sha256 = "sha256-XUgCVHnTLZXFU+r0s1yuYryWNBJRgQrFlf3g1iRrLWs=";
   };
 
   # bitmask-root is only used on GNU/Linux
@@ -105,7 +108,9 @@ buildGoModule rec {
 
   nativeBuildInputs = [
     cmake
+    git
     pkg-config
+    python3
     python3Packages.wrapPython
     which
     wrapQtAppsHook
@@ -130,6 +135,8 @@ buildGoModule rec {
   # gui/build.sh will build Go modules into lib/libgoshim.a
   buildPhase = ''
     runHook preBuild
+
+    make vendor
 
     # TODO: this is a hack that copies the qrc file that should by built by qmlcachegen
     # qmlcachegen is in qtdeclarative/libexec, but qmake is in qtbase/bin

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -3041,7 +3041,7 @@ with pkgs;
 
   bluetooth_battery = python3Packages.callPackage ../applications/misc/bluetooth_battery { };
 
-  calyx-vpn = libsForQt5.callPackage ../tools/networking/bitmask-vpn {
+  calyx-vpn = qt6Packages.callPackage ../tools/networking/bitmask-vpn {
     provider = "calyx";
     inherit (darwin.apple_sdk.frameworks) CoreFoundation Security;
   };

